src/context/internal.rs

```diff
@@ -52,7 +55,68 @@ pub struct ScopeMetadata {
 
 pub struct Scope<'a, R: Root + 'static> {
     pub metadata: ScopeMetadata,
-    pub handle_scope: &'a mut R
+    pub handle_scope: &'a mut R,
+    pub handle_arena: HandleArena<'a>
+}
+
+const CHUNK_SIZE: usize = 16;
+
+type Chunk = [raw::Local; CHUNK_SIZE];
+
+pub struct HandleArena<'a> {
+    chunks: VecDeque<Box<Chunk>>,
```

Comment:
`Vec` should be enough here I think?

Reply:
I think you're right: since these are arenas they only deallocate all at once, so `pop_front` will never be used.

Reply:
I ended up using an arena crate from crates.io